examples/jsm/loaders/OBJLoader.js

```diff
@@ -891,7 +891,7 @@ var OBJLoader = ( function () {
 
 					buffergeometry.setAttribute( 'position', new Float32BufferAttribute( state.vertices, 3 ) );
 
-					if ( state.colors.length > 0 ) {
+					if ( state.colors.length > 0 && state.colors[ 0 ] !== undefined ) {
```

Comment:
I see `undefined` is added to the colors array if a vertex does not declare them:

```js
state.colors.push( undefined, undefined, undefined );
```

What happens if only the first vertex has colors? Or there are _some_ colors in the middle of the file but not at the beginning? Wouldn't you want to add colors, then? Perhaps a flag should be set if colors are found in the file and color should otherwise default to `0` instead of `undefined`?

Reply:
I would prefer to keep it simple now and only do this if a user asks for it. I have not seen point cloud data with partial color definitions so far.

Reply:
I've already reported such an issue before in https://github.com/mrdoob/three.js/issues/16211 and this reintroduces the issue with inconsistent definitions albeit for point clouds.

Reply:
Like I said before, I've not seen point cloud OBJ with partial color definitions. That probably happens only in context of meshes.